src/index.js

```diff
@@ -1,16 +1,172 @@
+ *
+ * then add to package.json
+ *
+ *    "storage": {
+ *      "adapter": {
+ *        ...
+ *        "searcher": "waterline-elasticsearch"
+ *      },
+ *      "connections": {
+ *        ...
+ *        "searcher: {
+ *          "adapter": "searcher",
+ *          "host": "<host address>",
+ *          "port": 920,
+ *          "log": "warning",
+ *          "index": "searcher"
+ *        }
+ *      }
+ *    }
+ *
+ * ## Register model
+ * Now that the correct Storage adapters are configured, you'll need to tell Searcher which models you want to enable 
+ * search using the `searchable` method. Searchable accepts an identity for a model which has already been registered.
+ *
+ *    app.get('searcher').searchable('users')
+ *
+ * By default, Searcher will look for a field named `title` or `name` to use as the search field. You can specify different, or 
+ * multiple fields to search by specifying a second options parameter, with the `fields` key:
+ *
+ *    app.get('searcher').searchable('users', {fields: 'firstName'})
+ *    app.get('searcher').searchable('users', {fields: ['firstName', 'lastName']})
+ *
+ * ## Routes
+ * Based on the model identify, Searcher will create the following routes
+ *
+ *     /<model>/search
+ *
+ * which accepts a search parameter `q`. So to search for the term 'pizza':
+ *
+ *     /<model>/search?q=pizza
+ *
+ * The search wil return a list of results using the views below.
+ * 
+ * ## Views
+ * Searcher will automatically use the `@nxus/base-ui` views for any model that is searchable (if they exist).
+ *
+ * 1. view-<model>-list: the list view used to display search results.
+ * 1. view-<model>-detail: the detail view linked to from the list view.
+ *
+ * Alternatively, you can provide search specific views to be used instead:
+ *
+ * 1. search-<model>-list: the list view for returned search results.
+ * 1. search-<model>-detail: the detail view for an individual search result.
+ * 
+ * # API
+ * 
+ */
 
 'use strict';
 
+import SearchDocument from './models/searchDocument.js'
+import _ from 'underscore'
+import Promise from 'bluebird'
+
 /**
- * The Base Nxus Module
+ * The Search class enables automated searching of models using different adapters.
  */
-export default class Module {
+export default class Searcher {
   constructor(app) {
+    this.app = app
+    this.storage = this.app.get('storage')
+    this.router = this.app.get('router')
+    this.modelConfig = {}
+
+    this.app.get('searcher').use(this)
+    .gather('searchable')
+
+    this.storage.model(SearchDocument)
+  }
+
+  /**
+   * Register a model to be searchable.
+   * @param  {string} model the model identity
+   * @param  {Object} opts  An optional hash of options.
+   */
+  searchable(model, opts = {}) {
+    this.app.log.debug('Registering searchable', model, opts)
+    if(!opts.fields) opts.fields = ['name', 'title']
+    this.modelConfig[model] = opts
+
+    this.storage.on('model.create', this._handleCreate.bind(this))
```

Comment:
This would be a good place to use the model-specific `'model.'+model+'.create'` etc events rather than listening for all.